From: Jason Rhinelander <jason@imaginary.ca>
Date: Sun, 5 May 2019 01:27:17 -0300
Subject: Make oxen build and install shared libs

Patch the oxen cmake files to properly build and install shared libs.
---
 CMakeLists.txt          | 11 +++++++++++
 external/CMakeLists.txt |  6 ++++++
 src/CMakeLists.txt      |  1 -
 3 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ceb9a1..4f3c0bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -899,6 +899,17 @@ if(BUILD_DOCUMENTATION)
   endif()
 endif()
 
+set(oxenlibs epee blockchain_db checkpoints common cncrypto cryptonote_basic
+        cryptonote_core cryptonote_protocol device lmdb_lib mnemonics
+        multisig net p2p ringct_basic ringct sqlitedb
+        rpc_server_base rpc rpc_http_client rpc_commands daemon_rpc_server
+        wallet easylogging lmdb randomx blocks version)
+install(TARGETS ${oxenlibs}
+    ARCHIVE DESTINATION lib${OXEN_INSTALL_LIBDIR_SUFFIX}
+    LIBRARY DESTINATION lib${OXEN_INSTALL_LIBDIR_SUFFIX})
+set_target_properties(${oxenlibs} PROPERTIES
+    VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+
 find_package(PythonInterp)
 find_program(iwyu_tool_path NAMES iwyu_tool.py iwyu_tool)
 if (iwyu_tool_path AND PYTHONINTERP_FOUND)
diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index eddfdcf..89ace01 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -65,6 +65,12 @@ install(
     COMPONENT development
     FILES_MATCHING PATTERN "*.hpp")
 
+install(
+  DIRECTORY boost
+  DESTINATION include${OXEN_INSTALL_INCLUDEDIR_SUFFIX}
+  COMPONENT   development
+  FILES_MATCHING PATTERN "*.hpp")
+
 # uSockets doesn't really have a proper build system (just a very simple Makefile) so build it
 # ourselves.
 if (NOT CMAKE_VERSION VERSION_LESS 3.12)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a64130c..f79b934 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -81,7 +81,6 @@ endif()
 add_subdirectory(device)
 add_subdirectory(device_trezor)
 
-
 if(OXEN_VERSIONTAG)
   set(VERSIONTAG "${OXEN_VERSIONTAG}")
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/version.cpp.in" "${CMAKE_CURRENT_BINARY_DIR}/version.cpp")
